chi/server.py

```diff
@@ -85,211 +89,300 @@ def instance_create_args(
         # https://github.com/ChameleonCloud/horizon/blob/stable/liberty/openstack_dashboard/dashboards/project/instances/workflows/create_instance.py#L943
         # appears to POST a JSON akin to
         # {"server": {..., "networks": [{"uuid": "e8c33574-5423-436c-a45b-5bab78071b8a"}] ...}, "os:scheduler_hints": ...},
-        server_args["nics"] = [
-            {"net-id": netid, "v4-fixed-ip": ""} for netid in net_ids
+        server_args["networks"] = [
+            {"uuid": netid} for netid in net_ids
         ]
 
     server_args.update(kwargs)
     return server_args
 
+class Server:
+    def __init__(self, name: str, reservation_id: Optional[str] = None, image_name: str = DEFAULT_IMAGE_NAME,
+                 image: Optional[str] = None, flavor_name: str = BAREMETAL_FLAVOR,
+                 key_name: str = None, keypair: Optional[Keypair] = None,
+                 network_name: str = DEFAULT_NETWORK):
 
-class Server(object):
-    """A wrapper object referring to a server instance.
 
-    This class is helpful if you want to use a more object-oriented programming
-    approach when building your infrastrucutre. With the Server abstraction,
-    you can for example do the following:
+        self.name = name
+        self.reservation_id = reservation_id or None
+        # self.image = image or get_image(image_name)
+        self.image_name = image_name
+        self.flavor_name = flavor_name
 
-    .. code-block:: python
+        ## Add an update keypair call before this and make sure this does not crash on init
+        update_keypair()
```

Comment:
I don't think we are calling this correctly. Line 114 should handle this.
1. If the user passes in a keypair name, we should use that
2. Otherwise (upload if needed) and use the $USER-jupyter keypair